File: multi_agent/projects/im-desktop/services/groupManagementLogService.ts
import axios, { AxiosResponse } from 'axios';
import { GroupManagementLog, GroupManagementLogDTO, GroupManagementLogFilter, GroupManagementLogStatistics } from '../types/groupManagementLog';
import { apiClient, handleApiError } from './apiClient';

/**
 * 群管理日志服务
 */
export class GroupManagementLogService {
  private baseUrl = '/api/v1/group-management-logs';

  /**
   * 记录群管理操作日志
   */
  async logOperation(logDTO: GroupManagementLogDTO): Promise<GroupManagementLog> {
    try {
      const response: AxiosResponse<GroupManagementLog> = await apiClient.post(
        this.baseUrl,
        logDTO
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '记录群管理操作日志失败');
    }
  }

  /**
   * 批量记录操作日志
   */
  async batchLogOperations(logDTOs: GroupManagementLogDTO[]): Promise<GroupManagementLog[]> {
    try {
      const response: AxiosResponse<GroupManagementLog[]> = await apiClient.post(
        `${this.baseUrl}/batch`,
        logDTOs
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '批量记录群管理操作日志失败');
    }
  }

  /**
   * 根据ID查询日志
   */
  async getLogById(id: string): Promise<GroupManagementLog> {
    try {
      const response: AxiosResponse<GroupManagementLog> = await apiClient.get(
        `${this.baseUrl}/${id}`
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '查询日志失败');
    }
  }

  /**
   * 根据群组ID查询日志
   */
  async getLogsByGroupId(groupId: string): Promise<GroupManagementLog[]> {
    try {
      const response: AxiosResponse<GroupManagementLog[]> = await apiClient.get(
        `${this.baseUrl}/group/${groupId}`
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '查询群组日志失败');
    }
  }

  /**
   * 根据群组ID分页查询日志
   */
  async getLogsByGroupIdPage(
    groupId: string,
    page: number = 0,
    size: number = 20,
    sort: string = 'createdAt,desc'
  ): Promise<{ content: GroupManagementLog[]; totalElements: number; totalPages: number }> {
    try {
      const params = {
        page,
        size,
        sort,
      };
      const response: AxiosResponse<{
        content: GroupManagementLog[];
        totalElements: number;
        totalPages: number;
      }> = await apiClient.get(`${this.baseUrl}/group/${groupId}/page`, { params });
      return response.data;
    } catch (error) {
      throw handleApiError(error, '分页查询群组日志失败');
    }
  }

  /**
   * 根据操作者ID查询日志
   */
  async getLogsByOperatorId(operatorId: string): Promise<GroupManagementLog[]> {
    try {
      const response: AxiosResponse<GroupManagementLog[]> = await apiClient.get(
        `${this.baseUrl}/operator/${operatorId}`
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '查询操作者日志失败');
    }
  }

  /**
   * 根据目标用户ID查询日志
   */
  async getLogsByTargetUserId(targetUserId: string): Promise<GroupManagementLog[]> {
    try {
      const response: AxiosResponse<GroupManagementLog[]> = await apiClient.get(
        `${this.baseUrl}/target/${targetUserId}`
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '查询目标用户日志失败');
    }
  }

  /**
   * 根据操作类型查询日志
   */
  async getLogsByActionType(actionType: string): Promise<GroupManagementLog[]> {
    try {
      const response: AxiosResponse<GroupManagementLog[]> = await apiClient.get(
        `${this.baseUrl}/action/${actionType}`
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '查询操作类型日志失败');
    }
  }

  /**
   * 根据时间范围查询日志
   */
  async getLogsByTimeRange(startDate: Date, endDate: Date): Promise<GroupManagementLog[]> {
    try {
      const params = {
        start: startDate.toISOString(),
        end: endDate.toISOString(),
      };
      const response: AxiosResponse<GroupManagementLog[]> = await apiClient.get(
        `${this.baseUrl}/time-range`,
        { params }
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '查询时间范围日志失败');
    }
  }

  /**
   * 高级搜索日志
   */
  async searchLogs(
    filter: GroupManagementLogFilter,
    page: number = 0,
    size: number = 20,
    sort: string = 'createdAt,desc'
  ): Promise<{ content: GroupManagementLog[]; totalElements: number; totalPages: number }> {
    try {
      const params = {
        ...filter,
        page,
        size,
        sort,
      };
      const response: AxiosResponse<{
        content: GroupManagementLog[];
        totalElements: number;
        totalPages: number;
      }> = await apiClient.get(`${this.baseUrl}/search`, { params });
      return response.data;
    } catch (error) {
      throw handleApiError(error, '搜索日志失败');
    }
  }

  /**
   * 获取群组最近的操作日志
   */
  async getRecentLogsByGroupId(groupId: string, limit: number = 10): Promise<GroupManagementLog[]> {
    try {
      const params = { limit };
      const response: AxiosResponse<GroupManagementLog[]> = await apiClient.get(
        `${this.baseUrl}/group/${groupId}/recent`,
        { params }
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '获取最近日志失败');
    }
  }

  /**
   * 获取需要通知的日志
   */
  async getPendingNotificationLogs(): Promise<GroupManagementLog[]> {
    try {
      const response: AxiosResponse<GroupManagementLog[]> = await apiClient.get(
        `${this.baseUrl}/pending-notifications`
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '获取待通知日志失败');
    }
  }

  /**
   * 获取重要操作日志
   */
  async getImportantLogs(): Promise<GroupManagementLog[]> {
    try {
      const response: AxiosResponse<GroupManagementLog[]> = await apiClient.get(
        `${this.baseUrl}/important`
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '获取重要日志失败');
    }
  }

  /**
   * 标记日志为已通知
   */
  async markLogsAsNotified(logIds: string[]): Promise<number> {
    try {
      const response: AxiosResponse<number> = await apiClient.put(
        `${this.baseUrl}/mark-notified`,
        logIds
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '标记日志为已通知失败');
    }
  }

  /**
   * 获取操作统计信息
   */
  async getStatistics(): Promise<GroupManagementLogStatistics> {
    try {
      const response: AxiosResponse<GroupManagementLogStatistics> = await apiClient.get(
        `${this.baseUrl}/statistics`
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '获取统计信息失败');
    }
  }

  /**
   * 获取群组操作统计信息
   */
  async getStatisticsByGroupId(groupId: string): Promise<GroupManagementLogStatistics> {
    try {
      const response: AxiosResponse<GroupManagementLogStatistics> = await apiClient.get(
        `${this.baseUrl}/statistics/group/${groupId}`
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '获取群组统计信息失败');
    }
  }

  /**
   * 获取热门操作类型
   */
  async getTopActionTypes(): Promise<Array<{ actionType: string; count: number }>> {
    try {
      const response: AxiosResponse<Array<{ actionType: string; count: number }>> = await apiClient.get(
        `${this.baseUrl}/statistics/top-actions`
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '获取热门操作类型失败');
    }
  }

  /**
   * 获取活跃操作者
   */
  async getTopOperators(): Promise<Array<{ operatorId: string; count: number }>> {
    try {
      const response: AxiosResponse<Array<{ operatorId: string; count: number }>> = await apiClient.get(
        `${this.baseUrl}/statistics/top-operators`
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '获取活跃操作者失败');
    }
  }

  /**
   * 导出日志为CSV格式
   */
  async exportToCsv(
    groupId?: string,
    startDate?: Date,
    endDate?: Date
  ): Promise<Blob> {
    try {
      const params: any = {};
      if (groupId) params.groupId = groupId;
      if (startDate) params.startDate = startDate.toISOString();
      if (endDate) params.endDate = endDate.toISOString();

      const response: AxiosResponse<ArrayBuffer> = await apiClient.get(
        `${this.baseUrl}/export/csv`,
        {
          params,
          responseType: 'arraybuffer',
        }
      );

      return new Blob([response.data], { type: 'text/csv' });
    } catch (error) {
      throw handleApiError(error, '导出CSV失败');
    }
  }

  /**
   * 导出日志为JSON格式
   */
  async exportToJson(
    groupId?: string,
    startDate?: Date,
    endDate?: Date
  ): Promise<Blob> {
    try {
      const params: any = {};
      if (groupId) params.groupId = groupId;
      if (startDate) params.startDate = startDate.toISOString();
      if (endDate) params.endDate = endDate.toISOString();

      const response: AxiosResponse<ArrayBuffer> = await apiClient.get(
        `${this.baseUrl}/export/json`,
        {
          params,
          responseType: 'arraybuffer',
        }
      );

      return new Blob([response.data], { type: 'application/json' });
    } catch (error) {
      throw handleApiError(error, '导出JSON失败');
    }
  }

  /**
   * 批量归档日志
   */
  async archiveLogs(logIds: string[]): Promise<number> {
    try {
      const response: AxiosResponse<number> = await apiClient.put(
        `${this.baseUrl}/archive`,
        logIds
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '归档日志失败');
    }
  }

  /**
   * 清理已归档的旧日志
   */
  async cleanupArchivedLogs(cutoffDate: Date): Promise<number> {
    try {
      const params = { cutoffDate: cutoffDate.toISOString() };
      const response: AxiosResponse<number> = await apiClient.delete(
        `${this.baseUrl}/cleanup`,
        { params }
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '清理归档日志失败');
    }
  }

  /**
   * 检查重复操作
   */
  async checkDuplicateOperation(
    groupId: string,
    operatorId: string,
    actionType: string,
    targetUserId?: string,
    withinMinutes: number = 5
  ): Promise<boolean> {
    try {
      const params: any = {
        groupId,
        operatorId,
        actionType,
        withinMinutes,
      };
      if (targetUserId) params.targetUserId = targetUserId;

      const response: AxiosResponse<boolean> = await apiClient.get(
        `${this.baseUrl}/check-duplicate`,
        { params }
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '检查重复操作失败');
    }
  }

  /**
   * 健康检查
   */
  async healthCheck(): Promise<{ status: string; service: string; timestamp: string }> {
    try {
      const response: AxiosResponse<{ status: string; service: string; timestamp: string }> = await apiClient.get(
        `${this.baseUrl}/health`
      );
      return response.data;
    } catch (error) {
      throw handleApiError(error, '健康检查失败');
    }
  }

  /**
   * 下载文件
   */
  private downloadFile(blob: Blob, filename: string) {
    const url = window.URL.createObjectURL(blob);
    const link = document.createElement('a');
    link.href = url;
    link.download = filename;
    document.body.appendChild(link);
    link.click();
    document.body.removeChild(link);
    window.URL.revokeObjectURL(url);
  }

  /**
   * 导出并下载CSV文件
   */
  async exportAndDownloadCsv(
    groupId?: string,
    startDate?: Date,
    endDate?: Date,
    filename: string = 'group-management-logs.csv'
  ): Promise<void> {
    const blob = await this.exportToCsv(groupId, startDate, endDate);
    this.downloadFile(blob, filename);
  }

  /**
   * 导出并下载JSON文件
   */
  async exportAndDownloadJson(
    groupId?: string,
    startDate?: Date,
    endDate?: Date,
    filename: string = 'group-management-logs.json'
  ): Promise<void> {
    const blob = await this.exportToJson(groupId, startDate, endDate);
    this.downloadFile(blob, filename);
  }

  /**
   * 创建添加成员日志
   */
  createMemberAddLog(
    groupId: string,
    operatorId: string,
    targetUserId: string,
    operatorType: string = 'ADMIN',
    details: Record<string, any> = {}
  ): GroupManagementLogDTO {
    return {
      groupId,
      operatorId,
      operatorType,
      targetUserId,
      actionType: 'MEMBER_ADD',
      description: '添加新成员',
      details,
      result: 'SUCCESS',
      important: false,
      needNotification: true,
    };
  }

  /**
   * 创建移除成员日志
   */
  createMemberRemoveLog(
    groupId: string,
    operatorId: string,
    targetUserId: string,
    operatorType: string = 'ADMIN',
    reason?: string,
    details: Record<string, any> = {}
  ): GroupManagementLogDTO {
    return {
      groupId,
      operatorId,
      operatorType,
      targetUserId,
      actionType: 'MEMBER_REMOVE',
      description: `移除成员${reason ? ` (${reason})` : ''}`,
      details,
      result: 'SUCCESS',
      important: true,
      needNotification: true,
    };
  }

  /**
   * 创建角色变更日志
   */
  createRoleChangeLog(
    groupId: string,
    operatorId: string,
    targetUserId: string,
    operatorType: string = 'ADMIN',
    fromRole: string,
    toRole: string,
    details: Record<string, any> = {}
  ): GroupManagementLogDTO {
    return {
      groupId,
      operatorId,
      operatorType,
      targetUserId,
      actionType: 'ROLE_CHANGE',
      description: `角色变更: ${fromRole} → ${toRole}`,
      details,
      result: 'SUCCESS',
      important: true,
      needNotification: true,
    };
  }
}

// 导出单例实例
export const groupManagementLogService = new GroupManagementLogService();

File: multi_agent/projects/im-mobile/lib/services/group_management_log_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import '../models/group_management_log.dart';
import '../utils/api_client.dart';

/// 群管理日志服务
class GroupManagementLogService {
  final String baseUrl = '/api/v1/group-management-logs';

  /// 记录群管理操作日志
  Future<GroupManagementLog> logOperation(GroupManagementLogDTO logDTO) async {
    try {
      final response = await ApiClient.post(baseUrl, data: logDTO.toJson());
      return GroupManagementLog.fromJson(response.data);
    } catch (e) {
      throw ApiException('记录群管理操作日志失败: $e');
    }
  }

  /// 批量记录操作日志
  Future<List<GroupManagementLog>> batchLogOperations(
      List<GroupManagementLogDTO> logDTOs) async {
    try {
      final response = await ApiClient.post(
        '$baseUrl/batch',
        data: jsonEncode(logDTOs.map((e) => e.toJson()).toList()),
      );
      return (response.data as List)
          .map((e) => GroupManagementLog.fromJson(e))
          .toList();
    } catch (e) {
      throw ApiException('批量记录群管理操作日志失败: $e');
    }
  }

  /// 根据 ID 查询日志
  Future<GroupManagementLog> getLogById(String id) async {
    try {
      final response = await ApiClient.get('$baseUrl/$id');
      return GroupManagementLog.fromJson(response.data);
    } catch (e) {
      throw ApiException('查询日志失败: $e');
    }
  }

  /// 根据群组 ID 查询日志
  Future<List<GroupManagementLog>> getLogsByGroupId(String groupId) async {
    try {
      final response = await ApiClient.get('$baseUrl/group/$groupId');
      return (response.data as List)
          .map((e) => GroupManagementLog.fromJson(e))
          .toList();
    } catch (e) {
      throw ApiException('查询群组日志失败: $e');
    }
  }

  /// 根据群组 ID 分页查询日志
  Future<Map<String, dynamic>> getLogsByGroupIdPage(
    String groupId, {
    int page = 0,
    int size = 20,
    String sort = 'createdAt,desc',
  }) async {
    try {
      final response = await ApiClient.get(
        '$baseUrl/group/$groupId/page',
        queryParameters: {'page': page, 'size': size, 'sort': sort},
      );
      return {
        'content': (response.data['content'] as List)
            .map((e) => GroupManagementLog.fromJson(e))
            .toList(),
        'totalElements': response.data['totalElements'],
        'totalPages': response.data['totalPages'],
      };
    } catch (e) {
      throw ApiException('分页查询群组日志失败: $e');
    }
  }

  /// 根据操作者 ID 查询日志
  Future<List<GroupManagementLog>> getLogsByOperatorId(String operatorId) async {
    try {
      final response = await ApiClient.get('$baseUrl/operator/$operatorId');
      return (response.data as List)
          .map((e) => GroupManagementLog.fromJson(e))
          .toList();
    } catch (e) {
      throw ApiException('查询操作者日志失败: $e');
    }
  }

  /// 根据目标用户 ID 查询日志
  Future<List<GroupManagementLog>> getLogsByTargetUserId(String targetUserId) async {
    try {
      final response = await ApiClient.get('$baseUrl/target/$targetUserId');
      return (response.data as List)
          .map((e) => GroupManagementLog.fromJson(e))
          .toList();
    } catch (e) {
      throw ApiException('查询目标用户日志失败: $e');
    }
  }

  /// 根据操作类型查询日志
  Future<List<GroupManagementLog>> getLogsByActionType(String actionType) async {
    try {
      final response = await ApiClient.get('$baseUrl/action/$actionType');
      return (response.data as List)
          .map((e) => GroupManagementLog.fromJson(e))
          .toList();
    } catch (e) {
      throw ApiException('查询操作类型日志失败: $e');
    }
  }

  /// 根据时间范围查询日志
  Future<List<GroupManagementLog>> getLogsByTimeRange(
    DateTime startDate,
    DateTime endDate,
  ) async {
    try {
      final response = await ApiClient.get(
        '$baseUrl/time-range',
        queryParameters: {
          'start': startDate.toIso8601String(),
          'end': endDate.toIso8601String(),
        },
      );
      return (response.data as List)
          .map((e) => GroupManagementLog.fromJson(e))
          .toList();
    } catch (e) {
      throw ApiException('查询时间范围日志失败: $e');
    }
  }

  /// 高级搜索日志
  Future<Map<String, dynamic>> searchLogs({
    String? groupId,
    String? operatorId,
    String? targetUserId,
    String? actionType,
    String? actionSubType,
    String? result,
    bool? important,
    bool? needNotification,
    bool? notified,
    DateTime? startDate,
    DateTime? endDate,
    int page = 0,
    int size = 20,
    String sort = 'createdAt,desc',
  }) async {
    try {
      final params = <String, dynamic>{
        if (groupId != null) 'groupId': groupId,
        if (operatorId != null) 'operatorId': operatorId,
        if (targetUserId != null) 'targetUserId': targetUserId,
        if (actionType != null) 'actionType': actionType,
        if (actionSubType != null) 'actionSubType': actionSubType,
        if (result != null) 'result': result,
        if (important != null) 'important': important,
        if (needNotification != null) 'needNotification': needNotification,
        if (notified != null) 'notified': notified,
        if (startDate != null) 'startDate': startDate.toIso8601String(),
        if (endDate != null) 'endDate': endDate.toIso8601String(),
        'page': page,
        'size': size,
        'sort': sort,
      };

      final response = await ApiClient.get('$baseUrl/search', queryParameters: params);
      return {
        'content': (response.data['content'] as List)
            .map((e) => GroupManagementLog.fromJson(e))
            .toList(),
        'totalElements': response.data['totalElements'],
        'totalPages': response.data['totalPages'],
      };
    } catch (e) {
      throw ApiException('搜索日志失败: $e');
    }
  }

  /// 获取群组最近的操作日志
  Future<List<GroupManagementLog>> getRecentLogsByGroupId(
    String groupId, {
    int limit = 10,
  }) async {
    try {
      final response = await ApiClient.get(
        '$baseUrl/group/$groupId/recent',
        queryParameters: {'limit': limit},
      );
      return (response.data as List)
          .map((e) => GroupManagementLog.fromJson(e))
          .toList();
    } catch (e) {
      throw ApiException('获取最近日志失败: $e');
    }
  }

  /// 获取需要通知的日志
  Future<List<GroupManagementLog>> getPendingNotificationLogs() async {
    try {
      final response = await ApiClient.get('$baseUrl/pending-notifications');
      return (response.data as List)
          .map((e) => GroupManagementLog.fromJson(e))
          .toList();
    } catch (e) {
      throw ApiException('获取待通知日志失败: $e');
    }
  }

  /// 获取重要操作日志
  Future<List<GroupManagementLog>> getImportantLogs() async {
    try {
      final response = await ApiClient.get('$baseUrl/important');
      return (response.data as List)
          .map((e) => GroupManagementLog.fromJson(e))
          .toList();
    } catch (e) {
      throw ApiException('获取重要日志失败: $e');
    }
  }

  /// 标记日志为已通知
  Future<int> markLogsAsNotified(List<String> logIds) async {
    try {
      final response = await ApiClient.put(
        '$baseUrl/mark-notified',
        data: jsonEncode(logIds),
      );
      return response.data;
    } catch (e) {
      throw ApiException('标记日志为已通知失败: $e');
    }
  }

  /// 获取操作统计信息
  Future<GroupManagementLogStatistics> getStatistics() async {
    try {
      final response = await ApiClient.get('$baseUrl/statistics');
      return GroupManagementLogStatistics.fromJson(response.data);
    } catch (e) {
      throw ApiException('获取统计信息失败: $e');
    }
  }

  /// 获取群组操作统计信息
  Future<GroupManagementLogStatistics> getStatisticsByGroupId(String groupId) async {
    try {
      final response = await ApiClient.get('$baseUrl/statistics/group/$groupId');
      return GroupManagementLogStatistics.fromJson(response.data);
    } catch (e) {
      throw ApiException('获取群组统计信息失败: $e');
    }
  }

  /// 获取热门操作类型
  Future<List<Map<String, dynamic>>> getTopActionTypes() async {
    try {
      final response = await ApiClient.get('$baseUrl/statistics/top-actions');
      return (response.data as List).map((e) => Map<String, dynamic>.from(e)).toList();
    } catch (e) {
      throw ApiException('获取热门操作类型失败: $e');
    }
  }

  /// 获取活跃操作者
  Future<List<Map<String, dynamic>>> getTopOperators() async {
    try {
      final response = await ApiClient.get('$baseUrl/statistics/top-operators');
      return (response.data as List).map((e) => Map<String, dynamic>.from(e)).toList();
    } catch (e) {
      throw ApiException('获取活跃操作者失败: $e');
    }
  }

  /// 导出日志为 CSV 格式
  Future<Uint8List> exportToCsv({
    String? groupId,
    DateTime? startDate,
    DateTime? endDate,
  }) async {
    try {
      final params = <String, dynamic>{
        if (groupId != null) 'groupId': groupId,
        if (startDate != null) 'startDate': startDate.toIso8601String(),
        if (endDate != null) 'endDate': endDate.toIso8601String(),
      };

      final response = await ApiClient.get(
        '$baseUrl/export/csv',
        queryParameters: params,
        options: Options(responseType: ResponseType.bytes),
      );
      return response.data;
    } catch (e) {
      throw ApiException('导出 CSV 失败: $e');
    }
  }

  /// 导出日志为 JSON 格式
  Future<Uint8List> exportToJson({
    String? groupId,
    DateTime? startDate,
    DateTime? endDate,
  }) async {
    try {
      final params = <String, dynamic>{
        if (groupId != null) 'groupId': groupId,
        if (startDate != null) 'startDate': startDate.toIso8601String(),
        if (endDate != null) 'endDate': endDate.toIso8601String(),
      };

      final response = await ApiClient.get(
        '$baseUrl/export/json',
        queryParameters: params,
        options: Options(responseType: ResponseType.bytes),
      );
      return response.data;
    } catch (e) {
      throw ApiException('导出 JSON 失败: $e');
    }
  }

  /// 批量归档日志
  Future<int> archiveLogs(List<String> logIds) async {
    try {
      final response = await ApiClient.put(
        '$baseUrl/archive',
        data: jsonEncode(logIds),
      );
      return response.data;
    } catch (e) {
      throw ApiException('归档日志失败: $e');
    }
  }

  /// 清理已归档的旧日志
  Future<int> cleanupArchivedLogs(DateTime cutoffDate) async {
    try {
      final response = await ApiClient.delete(
        '$baseUrl/cleanup',
        queryParameters: {'cutoffDate': cutoffDate.toIso8601String()},
      );
      return response.data;
    } catch (e) {
      throw ApiException('清理归档日志失败: $e');
    }
  }

  /// 检查重复操作
  Future<bool> checkDuplicateOperation({
    required String groupId,
    required String operatorId,
    required String actionType,
    String? targetUserId,
    int withinMinutes = 5,
  }) async {
    try {
      final params = <String, dynamic>{
        'groupId': groupId,
        'operatorId': operatorId,
        'actionType': actionType,
        'withinMinutes': withinMinutes,
        if (targetUserId != null) 'targetUserId': targetUserId,
      };

      final response = await ApiClient.get(
        '$baseUrl/check-duplicate',
        queryParameters: params,
      );
      return response.data;
    } catch (e) {
      throw ApiException('检查重复操作失败: $e');
    }
  }

  /// 健康检查
  Future<Map<String, String>> healthCheck() async {
    try {
      final response = await ApiClient.get('$baseUrl/health');
      return Map<String, String>.from(response.data);
    } catch (e) {
      throw ApiException('健康检查失败: $e');
    }
  }

  /// 创建添加成员日志
  GroupManagementLogDTO createMemberAddLog({
    required String groupId,
    required String operatorId,
    required String targetUserId,
    String operatorType = 'ADMIN',
    Map<String, dynamic>? details,
  }) {
    return GroupManagementLogDTO(
      groupId: groupId,
      operatorId: operatorId,
      operatorType: operatorType,
      targetUserId: targetUserId,
      actionType: 'MEMBER_ADD',
      description: '添加新成员',
      details: details,
      result: 'SUCCESS',
      important: false,
      needNotification: true,
    );
  }

  /// 创建移除成员日志
  GroupManagementLogDTO createMemberRemoveLog({
    required String groupId,
    required String operatorId,
    required String targetUserId,
    String operatorType = 'ADMIN',
    String? reason,
    Map<String, dynamic>? details,
  }) {
    return GroupManagementLogDTO(
      groupId: groupId,
      operatorId: operatorId,
      operatorType: operatorType,
      targetUserId: targetUserId,
      actionType: 'MEMBER_REMOVE',
      description: '移除成员${reason != null ? ' ($reason)' : ''}',
      details: details,
      result: 'SUCCESS',
      important: true,
      needNotification: true,
    );
  }

  /// 创建角色变更日志
  GroupManagementLogDTO createRoleChangeLog({
    required String groupId,
    required String operatorId,
    required String targetUserId,
    String operatorType = 'ADMIN',
    required String fromRole,
    required String toRole,
    Map<String, dynamic>? details,
  }) {
    return GroupManagementLogDTO(
      groupId: groupId,
      operatorId: operatorId,
      operatorType: operatorType,
      targetUserId: targetUserId,
      actionType: 'ROLE_CHANGE',
      description: '角色变更：$fromRole → $toRole',
      details: details,
      result: 'SUCCESS',
      important: true,
      needNotification: true,
    );
  }
}